pantheon/src/main/java/tech/pegasys/pantheon/controller/PantheonControllerBuilderFactory.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.controller;
+
+import tech.pegasys.pantheon.cli.EthNetworkConfig;
+import tech.pegasys.pantheon.config.GenesisConfigFile;
+import tech.pegasys.pantheon.config.GenesisConfigOptions;
+
+public class PantheonControllerBuilderFactory {
+
+  public PantheonControllerBuilder<?> fromEthNetworkConfig(
+      final EthNetworkConfig ethNetworkConfig) {
+    return fromGenesisConfig(GenesisConfigFile.fromConfig(ethNetworkConfig.getGenesisConfig()))
+        .networkId(ethNetworkConfig.getNetworkId());
+  }
+
+  public PantheonControllerBuilder<?> fromGenesisConfig(final GenesisConfigFile genesisConfig) {
+    final GenesisConfigOptions configOptions = genesisConfig.getConfigOptions();
+    final PantheonControllerBuilder<?> builder;
+
+    if (configOptions.isEthHash()) {
```

Comment:
Not for this PR, but we should think of a way to make this pluggable without prior knowledge of all the controller builders.

Reply:
Absolutely agreed. I can see a few places that pluggability is starting to emerge and would make this setup code cleaner (creating the storage provider being a prime example).